docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml

```diff
@@ -225,110 +183,59 @@
-            {% if helm_chart_overrides is defined %}{{ helm_chart_overrides }}{% endif %}
-            {{ armada_pod.stdout }}:/tmp/.
-            -c tiller
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-
-      - name: Reinstall Deployment Manager (helmv2)
-        command: >-
-          /usr/local/sbin/helmv2-cli -- helm upgrade
-          --install deployment-manager
-          --values /tmp/{{ helm_chart_overrides | basename }}
-          /tmp/{{ manager_chart | basename }}
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-
-      when: helmv2_installed and helmv2_dm_exists.rc == 0
+    # DM system app helm check, upload/apply helm-chart
+    - block:  
+        # Find dm-system app helm-chart
+       - name: Find deployment-manager app helm-chart in application directory
+         find:
+          paths: /usr/local/share/applications/helm/
+          patterns: 'deployment-manager*.tgz'
+         register: app_helm_chart
+
+       - name: Fail deployment-manager app helm-chart does not exists
+         fail:
+           msg: "DM system app helm-chart does not exists in application directory"
+         when: (app_helm_chart.files[0].path | length == 0)
+   
+        # dm-system app upload/apply
+       - name: Upload deployment-manager app Helm Chart
+         shell: source /etc/platform/openrc; system application-upload {{ manager_app_chart }}
+
+       - name: Wait deployment-manager app to become uploaded
+         shell: source /etc/platform/openrc; system application-list --nowrap | grep deployment-manager | awk '{ print $10 }'
+         register: app_upload_status
+         retries: 10
+         delay: 6
+         until: ("uploaded" in app_upload_status.stdout)
+         ignore_errors: yes
+
+       - name: Fail if deployment-manager app upload fails
+         fail:
+           msg: "The upload of deployment-manager system app has failed.
+                 Expected 'uploaded' state but the app is in 
+                 '{{ app_upload_status.stdout }}' state "
+         when: ("upload-failed" in app_upload_status.stdout)
+
+       - name: Apply deployment-manager app
+         shell: source /etc/platform/openrc; system application-apply deployment-manager   
+         when: ("uploaded" in app_upload_status.stdout)
+
+       - name: Wait deployment-manager app to become applied
+         shell: source /etc/platform/openrc; system application-list --nowrap | grep deployment-manager | awk '{ print $10 }'
+         register: app_apply_status
+         retries: 30
+         delay: 10
+         until: ("applied" in app_apply_status.stdout)
+         ignore_errors: yes
+
+       - name: Fail if deployment-manager app apply fails
+         fail:
+           msg: "The apply of DM system app has failed. 
+                 Expected 'applied' state but the app is in 
+                 '{{ app_apply_status.stdout }}' state."
+         when: ("apply-failed" in app_apply_status.stdout)
```

Comment:
Fail if anything other than "applied".
when: not ("applied" in app_apply_status.stdout)